src/crepe/facade/SDLContext.h

```diff
@@ -79,19 +80,49 @@ class SDLContext {
 		KEYUP,
 		KEYDOWN,
 		SHUTDOWN,
-
+		WINDOW_MINIMIZE,
+		WINDOW_MAXIMIZE,
+		WINDOW_FOCUS_GAIN,
+		WINDOW_FOCUS_LOST,
+		WINDOW_MOVE,
+		WINDOW_RESIZE,
+		WINDOW_EXPOSE,
 	};
-	//! EventData struct for passing event data from facade
-	struct EventData {
-		SDLContext::EventType event_type = SDLContext::EventType::NONE;
+	struct KeyData {
 		Keycode key = Keycode::NONE;
 		bool key_repeat = false;
+	};
+	struct MouseData {
 		MouseButton mouse_button = MouseButton::NONE;
 		ivec2 mouse_position = {-1, -1};
 		int scroll_direction = -1;
 		float scroll_delta = INFINITY;
 		ivec2 rel_mouse_move = {-1, -1};
 	};
+	struct WindowData {
+		ivec2 move_delta;
+		ivec2 resize_dimension;
+	};
+	//! EventData struct for passing event data from facade
+	struct EventData {
+		SDLContext::EventType event_type = SDLContext::EventType::NONE;
+		KeyData key_data;
+		MouseData mouse_data;
+		WindowData window_data;
```

Comment:
Ik begrijp ook dat maar een van deze `_data` velden gevuld zal zijn per EventData instantie? Is dit niet geschikt voor een `union`?